.github/workflows/test.yml

```diff
@@ -64,23 +64,23 @@ jobs:
 
  tests:
     name: Python ${{ matrix.python-version }}
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
 
     strategy:
       fail-fast: false
       max-parallel: 5
       matrix:
         python-version:
-          - 3.7
-          - 3.8
-          - 3.9
+          - '3.7'
+          - '3.8'
+          - '3.9'
           - '3.10'
-          - '3.11-dev'
+          - '3.11'
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v5
```

Comment:
can you please make the changes to CI in a separate commit or PR? the rationale is to keep the commit history clean, so that it makes sense to look back at a change set for the future us.